stablehlo/dialect/StablehloOps.td

```diff
@@ -2788,12 +2719,6 @@ def StableHLO_DynamicGatherOp: StableHLO_Op<"dynamic_gather",
     DefaultValuedAttr<BoolAttr, "false">:$indices_are_sorted
   );
   let results = (outs HLO_Tensor);
-
-  let extraClassDeclaration = [{
-    static bool isCompatibleReturnTypes(TypeRange l, TypeRange r) {
-      return succeeded(mlir::verifyCompatibleShapes(l, r));
-    }
-  }];
 }
 
 def StableHLO_DynamicConvOp : StableHLO_Op<"dynamic_conv", [NoSideEffect]> {
```

Comment:
What kind of tests do we want to add to test this new functionality?

Reply:
Not now: same as mentioned in another comment, this PR loose the checks (inferred type VS real return type), so that all tests are expected to pass without any changes. When we implement dynamism one by one for each op, finer change specific for each op can be done case by case only when necessary.